### ISSUE TITLE
The highligh is not cleared when accepting a suggestion

### ISSUE THREAD

**Issue by korken89** [zsh-users/zsh-autosuggestions#252]
I am not sure if I am doing something wrong, but I am unable to find others having this issue, perhaps I have stumbled on some special case?
Anyways, I hope this clears the issue. :)

- I am running a clean install of zsh & oh-my-zsh with the following added lines in a urxvt-unicode-256color:
ZSH_THEME="agnoster"
plugins=(git ssh-agent zsh-autosuggestions)
RPROMPT='%F{241}[$(date +%H:%M:%S)]%f'
ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE='fg=030'

- zsh version is 5.1.1

When accepting a suggestion (with arrow key), the hightligh is not cleared and anything written after is in the color of the highlight.

See the screen shot for example:
![image](https://cloud.githubusercontent.com/assets/913109/26523896/126b27f2-4323-11e7-95b3-2593568dc831.png)

**Comment by stevenschmatz** [comment]
I'm getting the same issue here!

**Comment by michaelorr** [comment]
I was having this same problem and figured out the issue.
zle doesn't play nice with the leading zeros in the format string.
I was seeing the bug with `ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE='fg=008'` but the behavior is correct if I set `ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE='fg=8'`

Similarly I tested with `ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE='fg=030'` as you have and can see the buggy behavior, but `ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE='fg=30'` behaves properly.

Maybe a note in the readme or docs about properly formatting zle strings would be helpful, but this doesn't look like an issue with zsh-autosuggestions.
